Microservices assignment

germanleonz opened the pull request.

Commit:
Initial commit

Commit:
- Added support for Products.
- Added Services instead of accessing repositories directly from the
Controllers

Commit:
Removed unnecessary Constructors and Setters

Commit:
Added relationships between tables by using the corresponding
annotations

Commit:
Added Stocks

Commit:
Temporary commit

Commit:
Added Offers

Commit:
Added deletion of Products and Stores

Commit:
Removing dotfiles from the repo

Commit:
Removing unused files and code

Commit:
Added Stock checks on order creation

shuangli commented on src/main/java/com/tenx/ms/retail/stock/repository/StockRepository.java:
```diff
@@ -0,0 +1,13 @@
+
+@Repository
+public interface StockRepository extends JpaRepository<StockEntity, Long>{
+    Optional<StockEntity> findOneByStore_StoreIdAndProductId(Long storeId, Long productId);
```
It is not recommended to use _ in method name. for this case, you can change it to findOneByStoreIdAndProductId

germanleonz replied:
@shuangli thanks. Fixed.

shuangli commented:
please remove lib/spring-data-jpa-1.3.3.jar

Commit:
Removing lib/spring-data-jpa-1.3.3.jar

Commit:
Removed underscores from the Product and Stock repositories methods
names

shuangli commented:
where are your test cases?

germanleonz commented:
@shuangli I'm currently working on them. I'll upload them tomorrow.

wingng commented on src/main/java/com/tenx/ms/retail/order/domain/OrderEntity.java:
```diff
@@ -0,0 +1,119 @@
+    @Column(name = "order_date", columnDefinition = "datetime default current_timestamp",
+            insertable = false, nullable = false)
+    @Temporal(TemporalType.TIMESTAMP)
+    private Date orderDate;
```
You can use LocalDateTime from Java 8. It's better version of Date class

Commit:
- Added tests for the Store, Product and Stock controllers
- Removed the spring-data-rest-webmvc artifact from the POM file
- Replaced the ResourceNotFoundException with the NoSuchElementException

Commit:
Created a RequestHandler class to hold the code common between all of
the controller's tests

Commit:
- Added the OrderControllerTest
- Changed the tests so now the request are based on JSON files that
represent the RequestBody
